### DISCUSSION TITLE
MAC OS (Intel) agent cant get is to run as a service

### DISCUSSION THREAD

**Discussion by WJMIES** [microsoft/azure-pipelines-agent#4395] [Q&A]
Hi, I am having issues with a rather old Mac mini (don't ask :) ) we cant get the agent to run as a service.
mac is using Monterey.

We are using on prem Azure Devops 2022 server I have tried the version shipped with the server & 3.220.5 from this page am I doing some thin wrong?

**Error:**
`mac:myagent mte.build$ ./svc.sh start
starting vsts.agent.azdev.Engines_Mac_intel.mac
Failed: failed to load /Users/mte.build/Library/LaunchAgents/vsts.agent.azdev.Engines_Mac_intel.mac.plist
`
The plist file is there and was generated from the config.sh file using unattended we can run it interactively using run.sh but this isn't ideal

**Installation Steps:**

`cd ~/Downloads/
wget https://vstsagentpackage.azureedge.net/agent/3.220.5/vsts-agent-osx-x64-3.220.5.tar.gz
cd ~/
mkdir myagent 
cd myagent
tar xzf ~/Downloads/vsts-agent-osx-x64-3.220.5.tar.gz
./config.sh --unattended --url <devOps instance> --auth pat --token <insert Token> --pool "<insert Pool name>" 
sudo ./svc.sh install
sudo ./svc.sh start
sudo ./svc.sh status`

**Comment by WJMIES** [comment]
Update,
I have found that if we leave the "Desktop" session logged on the service runs and stays active as long as the "desktop" is logged on, obviously this is no an ideal situation to leave a machine even a MAC in a logged on state.

**Comment by petgus-lyko** [comment]
Having the same issue, also on a Mac mini
Install went fine
```
admin@machine azure-agent % ./svc.sh install
Creating launch agent in /Users/admin/Library/LaunchAgents/vsts.agent.XXXXGroup.Mac-Mini-Pool.xxxxxxx.plist
Creating /Users/admin/Library/Logs/vsts.agent.XXXXGroup.Mac-Mini-Pool.xxxxxxx
Creating /Users/admin/Library/LaunchAgents/vsts.agent.XXXXGroup.Mac-Mini-Pool.xxxxxxx.plist
Creating runsvc.sh
Creating .service
svc install complete
```

But then trying to start results in the error
```
admin@machine azure-agent % ./svc.sh start
starting vsts.agent.XXXXGroup.Mac-Mini-Pool.xxxxxxx
Failed: failed to load /Users/admin/Library/LaunchAgents/vsts.agent.XXXXGroup.Mac-Mini-Pool.xxxxxxx.plist
```

**Reply by petgus-lyko** [reply]
I solved it! 
`launchctl` will not load plist when connected via SSH terminal.

See this thread
https://github.com/microsoft/azure-pipelines-agent/issues/4498
